### PR TITLE
feat: migrate cli scripts and their test cases

### PR DESCRIPTION
**Pull request by PaperStrike**
Resolve https://github.com/MichaelDeBoey/eslint-find-rules/pull/13#discussion_r744206525

This adapts the bin scripts and their test files to the asynchronous API in https://github.com/MichaelDeBoey/eslint-find-rules/pull/13.

### CODE REVIEW

**Comment by PaperStrike** [comment]
Parent: https://github.com/sarbbottam/eslint-find-rules/pull/337

File: src/bin/diff.js [hunk 1]
```diff
@@ -94,3 +97,10 @@ function rulesDifference(a, b) {
     )
   );
 }
+
+process.exit(0);
+
+})().catch((e) => {
```
Comment (PaperStrike) [inline src/bin/diff.js:103]: @ljharb Should we test this? Or just ignore it? 🤔 E.g.,

```suggestion
})().catch(/* istanbul ignore next */(e) => {
```

Comment (ljharb) [reply src/bin/diff.js:103]: i think it's fine to ignore it.

**Comment by ljharb** [comment]
This needs to be rebased, which will hopefully resolve the test failures.

File: src/bin/diff.js (hunk 1 above)
Comment (PaperStrike) [reply src/bin/diff.js]: Done!

**Pull request merged**